lib/path.js

```diff
@@ -562,8 +582,38 @@ const win32 = {
     if (from === to)
       return '';
 
-    var fromOrig = win32.resolve(from);
-    var toOrig = win32.resolve(to);
+    var fromOrig;
+    var toOrig;
+    var isFromAbsolute = true;
+    var isToAbsolute = true;
+
+    try {
+      fromOrig = win32.resolve(from);
+    } catch (err) {
+      if (err.code === 'ERR_UNSUPPORTED_PLATFORM')
+        isFromAbsolute = false;
+    }
+    try {
+      toOrig = win32.resolve(to);
+    } catch (err) {
+      if (err.code === 'ERR_UNSUPPORTED_PLATFORM')
+        isToAbsolute = false;
+    }
+
+    if (process.platform !== 'win32') {
+      if (!isFromAbsolute && !isToAbsolute) {
```

Comment:
Because need to know both type of `from` and `to`, so I think I can't do this separately.